src/site/antora/modules/ROOT/pages/manual/appenders/database.adoc

```diff
@@ -0,0 +1,1505 @@
+
+| [[MongoDb4Provider-attr-connection]]connection
+| https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-core/com/mongodb/ConnectionString.html[`ConnectionString`]
+|
+|
+It specifies the connection URI used to reach the server.
+
+See
+https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/connection/connect/#connection-uri[Connection URI documentation]
+for its format.
+
+**Required**
+
+| [[MongoDb4Provider-attr-capped]]capped
+| `boolean`
+| `false`
+|
+If `true`, a
+https://www.mongodb.com/docs/manual/core/capped-collections/[capped collection]
+will be used.
+
+| [[MongoDb4Provider-attr-collectionSize]]collectionSize
+| `long`
+| `512 MiB`
+|
+It specifies the capped collection size of bytes.
+
+|===
+
+Additional runtime dependencies are required to use the `MongoDb4` provider:
+
+[tabs]
+====
+Maven::
++
+[source,xml]
+----
+<!-- We assume you use `log4j-bom` for dependency management -->
+<dependency>
+  <groupId>org.apache.logging.log4j</groupId>
+  <artifactId>log4j-mongodb4</artifactId>
+  <scope>runtime</scope>
+</dependency>
+----
+
+Gradle::
++
+[source,groovy]
+----
+// We assume you use `log4j-bom` for dependency management
+runtimeOnly 'org.apache.logging.log4j:log4j-mongodb4'
+----
+====
+
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-mongodb4_org-apache-logging-log4j-mongodb4-MongoDb4Provider[{open-book} Plugin reference for `MongoDb4`]
+
+[#CouchDbProvider]
+=== Apache CouchDB provider
+
+The `CouchDb` Provider allows using the <<NoSqlAppender>> with an
+https://couchdb.apache.org/[Apache CouchDB database].
+The provider can be configured by:
+
+* either providing some <<CouchDbProvider-attr-standard,standard configuration attributes>>,
+* or providing a <<CouchDbProvider-attr-factory,factory method>>.
+
+[#CouchDbProvider-attributes]
+.CouchDb provider configuration attributes
+[cols="1m,1,2,5"]
+|===
+| Attribute | Type | Default value | Description
+
+
+4+h| [[CouchDbProvider-attr-standard]]Standard configuration attributes
+
+| [[CouchDbProvider-attr-protocol]]protocol
+| _enumeration_
+| `http`
+a|
+It specifies the protocol to use to connect to the server.
+Can be one of:
+
+* `http`
+* `https`
+
+| [[CouchDbProvider-attr-server]]server
+| `String`
+| `localhost`
+|
+The host name of the CouchDB server.
+
+| [[CouchDbProvider-attr-port]]port
+| `int`
+a|
+`80`:: for `http`
+`443`:: for `https`
+|
+It specifies the TCP port to use.
+
+| [[CouchDbProvider-attr-databaseName]]databaseName
+| `String`
+|
+|
+The name of the database to connect to.
+
+| [[CouchDbProvider-attr-username]]username
+| `String`
+|
+|
+The username for authentication.
+
+| [[CouchDbProvider-attr-password]]password
+| `String`
+|
+|
+The password for authentication.
+
+4+h| [[CouchDbProvider-attr-factory]]Factory method configuration attributes
+
+| [[CouchDbProvider-attr-factoryClassName]]factoryClassName
+| https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html[`Class<?>`]
```

Comment:
Mind removing Javadoc links to basic standard classes like `Class`, please?

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2790/commits/79d21ae2a4321c7bc516a9fb99689cf7059c1603